js/wordsearch.js

```diff
@@ -0,0 +1,346 @@
+(function(){
+  'use strict';
+
+ // Initialize the application
+  WordSearch.prototype.initialize = function() {
+    this.matrix      = [];                            // Letter matrix
+    this.selectFrom  = null;                         // Selection from
+    this.currentWord = [];                          // currentWord
+    this.initmatrix(this.settings.gridSize,this.matrix);
+  }
+
+   /**
+   * Fill default items into the matrix
+   * @param {Number} Grid size
+   * @param {array} Grid matrix
+   */
+  WordSearch.prototype.initmatrix = function(size,matrix) {
+    for (var row = 0; row < size; row++) {
+      for (var col = 0; col < size; col++) {
+        var item = {
+          letter: '.',                                 // Default value
+          row: row,
+          col: col
+        }
+        if (!this.matrix[row]) this.matrix[row] = [];
+        this.matrix[row][col] = item;
+      }
+    }
+  }
+
+  /**
+   * Word seach
+   * @param {Element} context gridSection of game
+   * @param {Array} settings
+   * @constructor
+   */
+  function WordSearch(context,settings) {
+    this.hintFound = 0;                                             // hint word found
+    this.defaultSettings = {                                        // Default settings
+      'directions'  : ['W', 'N', 'WN', 'EN'],
+      'gridSize'    : 10,
+      'words'       : ['one', 'two', 'three', 'four', 'five'],
+      'wordsList'   : [],
+      'fillLetters' : false
+    }
+    this.gridSection = context;
+    this.gridSection.classList.add('word-search');                      // Add `.word-search` to wrap element
+    this.settings = {...this.settings, ...this.defaultSettings};
```

Comment:
there you are overriding the current setting with default one, which is not expected